### PR TITLE
Hotfixes for Directory and PUT as PATCH

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: lib/Controller/DirectoryController.php [hunk 1]
```diff
@@ -75,25 +76,18 @@ public function update(): JSONResponse
...
-		}
-		
-		// Validate the URL
-		if (!filter_var($url, FILTER_VALIDATE_URL)) {
-			return new JSONResponse(['error' => 'Invalid URL provided'], 400);
-		}	
 
 		// Sync the external directory with the provided URL
-		$data = $this->directoryService->syncExternalDirectory($url);
+		try{
```
Comment (WilcoLouwerse) [inline lib/Controller/DirectoryController.php]: try {

File: lib/Controller/ListingsController.php [hunk 2]
```diff
@@ -186,13 +187,16 @@ public function add(): JSONResponse
...
         $url = $this->request->getParam('url');
 
-		// Check if the URL parameter is provided
-		if (empty($url) === true) {
-			return new JSONResponse(['error' => 'URL parameter is required'], 400);
-		}
-
         // Add the new listing using the provided URL
-        $result = $this->directoryService->syncExternalDirectory($url);
+		try{
```
Comment (WilcoLouwerse) [inline lib/Controller/ListingsController.php]: try {

File: lib/Service/DirectoryService.php [hunk 3]
```diff
@@ -342,11 +390,13 @@ public function syncExternalDirectory(string $url): array
 		// array_column() with null as second parameter returns complete array entries
 		// This will return complete listing objects indexed by their catalog ID
 		$oldListings = array_column(
-			$currentListings, 
+			$currentListings,
 			null, // null returns complete array entries rather than a specific column
 			'catalog' // Index by catalog ID
 		);
 
+		var_dump($oldListings, $currentListings);
```
Comment (bbrands02) [inline lib/Service/DirectoryService.php:398]: dump

File: lib/Service/DirectoryService.php [hunk 4]
```diff
@@ -365,10 +415,12 @@ public function syncExternalDirectory(string $url): array
 			// Check if we already have this listing by looking up its catalog ID in the oldListings array
 			$oldListing = $oldListings[$listing['id']] ?? null;
 
+			var_dump($listing['id'], $oldListing, $oldListings);
```
Comment (bbrands02) [inline lib/Service/DirectoryService.php:418]: dump

**Review by bbrands02 — APPROVED** [review]

File: lib/Controller/DirectoryController.php [hunk 5]
```diff
@@ -75,25 +76,18 @@ public function update(): JSONResponse
...
 		// Sync the external directory with the provided URL
-		$data = $this->directoryService->syncExternalDirectory($url);
+		try {
+			$data = $this->directoryService->syncExternalDirectory($url);
+		} catch (DirectoryUrlException $exception) {
+			if($exception->getMessage() === 'URL is required') {
+				$exception->setMessage('Property "directory" is required');
+			}
+
+
```
Comment (WilcoLouwerse) [inline lib/Controller/DirectoryController.php:88]: newlines

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**